src/ring/rack.clj

```diff
@@ -25,8 +25,9 @@
 
       def responsify(output)
         begin
-          return output[0].to_java if output.respond_to?(:size) && output.size > 1
-          output.each{|s| s.to_java}
+          retval = []
+          output.each {|s| retval.push(s)}
+          retval
```

Comment:
`each` is not lazy, but it doesn't change the object itself and returns the original object, so only side effects.

With the original code only HTML pages work, assets (images, generated javascript) not.